### PR TITLE
[fsdp] fix: avoid NestedTensor jagged dim ambiguity for 3D position_ids

### PR DESCRIPTION
**Pull request by Solus-sano**
### What does this PR do?

Fix an intermittent shape mismatch error during VLM SFT training with FSDP when using `DatasetPadMode.NO_PADDING`. The error manifests as:
```
The size of tensor a (3) must match the size of tensor b (8)
```

The root cause is a PyTorch `NestedTensor` jagged layout ambiguity: when all samples in a micro-batch have the same `seq_len`, `torch.nested.as_nested_tensor` incorrectly selects the first non-batch dimension (e.g., `num_heads=4`) as the jagged dimension instead of the `seq_len` dimension. This causes `.values()` to return shape `(B*num_heads, seq_len)` instead of `(num_heads, total_nnz)`.

### Checklist Before Starting

- [x] Search for similar PRs: [search link](https://github.com/verl-project/verl/pulls?q=is%3Apr+nested+tensor+position_ids)
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

Verified with the following test cases:

| Scenario | Original `.values()` | Fixed `unbind+cat` |
|----------|---------------------|-------------------|
| Different seq_len (2 samples) | `(4, 1, 300)` ✅ | `(4, 1, 300)` ✅ |
| Same seq_len (2 samples) | `(8, 1, 100)` ❌ | `(4, 1, 200)` ✅ |
| Same seq_len (8 samples) | `(32, 1, 100)` ❌ | `(4, 1, 800)` ✅ |

<details>
<summary>Reproduction script</summary>

```python
import torch

def fixed_position_ids_rmpad(position_ids_nt):
    return torch.cat([t for t in position_ids_nt.unbind()], dim=-1).unsqueeze(1)

def original_position_ids_rmpad(position_ids_nt):
    return position_ids_nt.values().unsqueeze(1)

# Case 1: Different seq_len - both methods agree
t1 = torch.arange(4*100).reshape(4, 100).float()
t2 = torch.arange(4*200).reshape(4, 200).float()
nt = torch.nested.as_nested_tensor([t1, t2], layout=torch.jagged)
assert original_position_ids_rmpad(nt).shape == fixed_position_ids_rmpad(nt).shape == (4, 1, 300)

# Case 2: Same seq_len - original method fails
t1 = torch.arange(4*100).reshape(4, 100).float()
t2 = torch.arange(4*100, 4*200).reshape(4, 100).float()
nt = torch.nested.as_nested_tensor([t1, t2], layout=torch.jagged)
assert original_position_ids_rmpad(nt).shape == (8, 1, 100)  # WRONG
assert fixed_position_ids_rmpad(nt).shape == (4, 1, 200)     # CORRECT
```

</details>

### API and Usage Example

No API changes. This is an internal fix.

### Design & Code Changes

Replace `.values()` with `unbind()` + `cat()` for 3D `position_ids` (shape `(B, num_heads, seq_len)`) in `FSDPEngineWithLMHead`:

```python
# Before (buggy when all seq_len are equal):
position_ids_rmpad = position_ids.values().unsqueeze(1)

# After (always correct):
position_ids_rmpad = torch.cat([t for t in position_ids.unbind()], dim=-1)
position_ids_rmpad = position_ids_rmpad.unsqueeze(1)
```

`unbind()` decomposes the NestedTensor along `dim=0` (batch) back into individual regular tensors of shape `(num_heads, seq_len_i)`, then `cat(..., dim=-1)` concatenates along the last dimension. This bypasses the jagged layout ambiguity entirely.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs). (N/A - internal bug fix, no doc changes needed)
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: This is a data-dependent edge case triggered only when all samples in a dynamic micro-batch have identical sequence lengths. A unit test would require mocking the batching pipeline.
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1).

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by wuxibin89** [comment]
We should use `nested_tensor_from_jagged` instead of `nested_tensor` to process 3d postition_ids.
https://github.com/verl-project/verl/blob/main/verl/utils/dataset/dataset_utils.py#L70
```
import torch
x = torch.rand(4, 10)
y = torch.rand(4, 10)

nt = torch.nested.nested_tensor([x, y], layout=torch.jagged)
print(nt.values().shape) # incorrect: torch.Size([8, 10])

values = torch.cat([x, y], dim=1)
offsets = torch.tensor([0, 10])
nt = torch.nested.nested_tensor_from_jagged(values, offsets, jagged_dim=1)
print(nt.values().shape) # correct: torch.Size([4, 20])
```
